src/utils.js

```diff
@@ -36,38 +34,39 @@ function throttleAll(limit, tasks) {
   }
 
   return new Promise((resolve, reject) => {
-    // eslint-disable-next-line unicorn/new-for-builtins
-    const result = Array(tasks.length).fill(notSettled);
+    const result = /** @type {T[]} */ ([]);
     const entries = tasks.entries();
+    let tasksFulfilled = 0;
+
     const next = () => {
       const { done, value } = entries.next();
 
-      if (done) {
-        const isLast = !result.includes(notSettled);
-
-        if (isLast) {
-          // eslint-disable-next-line node/callback-return
-          resolve(result);
-        }
+      if (tasksFulfilled === tasks.length) {
```

Comment:
I think we can move it inside `done` as is was, no need to check it on the each call

Reply:
ok, done